common/lib/aws_client.ts

```diff
@@ -58,6 +58,8 @@ export abstract class AwsClient extends EventEmitter {
 
     this.properties = new Map<string, any>(Object.entries(config));
 
+    driverDialect.setupDriverProperties(this.properties);
```

Comment:
Is there a reason this is called so early before we actually use it?
Do we really need this extra public facing method? 
What do you think of setting up the connect timeout directly  within `DriverDialect#connect`
```
async connect(hostInfo: HostInfo, props: Map<string, any>): Promise<ClientWrapper> {
    const wrapperConnectTimeout = props.get(WrapperProperties.WRAPPER_CONNECT_TIMEOUT.name);

    if (wrapperConnectTimeout) {
      props.set(this.connectTimeoutPropertyName, wrapperConnectTimeout);
    }
    const targetClient = await createConnection(WrapperProperties.removeWrapperProperties(props));
    return Promise.resolve(new MySQLClientWrapper(targetClient, hostInfo, props, this));
  }
```

Reply:
On second thought, I am okay with keeping the code contained as a method if it looks cleaner.
I think we should separate them into individual methods like setConnectTimeout and setQueryTimeout, and call them before we connect or query, so the intent is more clear.

Reply:
What do you think @sergiyvamz

Reply:
This is called here in order to set up the properties with the correct timeout values prior to creating objects like the telemetryFactory, pluginService and pluginManager, all of which accept a copy of the properties.

Reply:
I don't think it's important to setup driver specific properties before creating mentioned objects. These objects (telemetry factory, plugin service, etc.) are not suppose to access such driver specific properties and they don't know about them.

Reply:
I agree with Karen's idea to move this code to driver dialect connect().